hooks/hiero_upload_thumbnail.py

```diff
@@ -62,4 +63,12 @@ def execute(self, entity, source, item, **kwargs):
             tb = traceback.format_exc()
             self.parent.log_debug(tb)
         finally:
-            shutil.rmtree(thumbdir)
+            # Sometimes Windows holds on to the temporary thumbnail file longer than expected which
+            # can cause an exception here. If we wait a second and try again, this usually solves
+            # the issue.
+            try:
+                shutil.rmtree(thumbdir)
+            except:
```

Comment:
This should be an:

`except Exception:`

The reason being that a bare except will catch things that are best left uncaught, like keyboard interrupts and sig terms from the OS, while Exception acts as the base class for all "safe" exceptions to handle.